internal/pkg/v1/action/action_suite_test.go

```diff
@@ -7,46 +7,85 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/interceptor"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	olmv1 "github.com/operator-framework/operator-controller/api/v1"
 )
 
+const (
+	verbCreate = "create"
+)
+
 func TestCommand(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "Internal v1 action Suite")
 }
 
-type mockCreator struct {
+type fakeClient struct {
 	createErr    error
-	createCalled int
-}
-
-func (mc *mockCreator) Create(ctx context.Context, obj client.Object, opts ...client.CreateOption) error {
-	mc.createCalled++
-	return mc.createErr
-}
-
-type mockDeleter struct {
 	deleteErr    error
+	getErr       error
+	createCalled int
 	deleteCalled int
+	getCalled    int
+	transformers []objectTransformer
+	client.Client
 }
 
-func (md *mockDeleter) Delete(ctx context.Context, obj client.Object, opts ...client.DeleteOption) error {
-	md.deleteCalled++
-	return md.deleteErr
+// meant to apply a function to anything matching the objectKey
+// when the fakeClient an action corresponding to verb.
+// transformer will not run if an error is already specified
+// for the verb in the fakeClient.
+// only implemented in Create for now
+type objectTransformer struct {
+	verb          string
+	objectKey     client.ObjectKey
+	transformFunc func(obj *client.Object)
 }
 
-type mockGetter struct {
-	getErr    error
-	getCalled int
-}
-
-func (mg *mockGetter) Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
-	mg.getCalled++
-	return mg.getErr
+func (c *fakeClient) Initialize() error {
+	scheme, err := action.NewScheme()
+	if err != nil {
+		return err
+	}
+	clientBuilder := fake.NewClientBuilder().WithInterceptorFuncs(interceptor.Funcs{
```

Comment:
that's cool, knowing about `WithInterceptorFuncs` and `"sigs.k8s.io/controller-runtime/pkg/client/interceptor"` definitely makes writing unit tests easier ;)